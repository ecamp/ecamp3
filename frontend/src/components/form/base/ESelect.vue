<template>
  <ValidationProvider
    v-slot="{ errors: veeErrors }"
    tag="div"
    :name="name"
    :vid="veeId"
    :rules="veeRules"
    :required="required"
    class="e-form-container"
  >
    <v-select
      v-bind="$attrs"
      :filled="filled"
      :hide-details="hideDetails"
      :error-messages="veeErrors.concat(errorMessages)"
      :label="label || name"
      :class="[inputClass]"
      v-on="$listeners"
    >
      <!-- passing through all slots -->
      <slot v-for="(_, name) in $slots" :slot="name" :name="name" />
      <template v-for="(_, name) in $scopedSlots" :slot="name" slot-scope="slotData">
        <slot :name="name" v-bind="slotData" />
      </template>
    </v-select>
  </ValidationProvider>
</template>

<script>
import { ValidationProvider } from 'vee-validate'
import { formComponentPropsMixin } from '@/mixins/formComponentPropsMixin.js'
import { formComponentMixin } from '@/mixins/formComponentMixin.js'

export default {
  name: 'ESelect',
  components: { ValidationProvider },
  mixins: [formComponentPropsMixin, formComponentMixin],
}
</script>
