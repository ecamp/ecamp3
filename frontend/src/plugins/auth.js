import axios from 'axios'
import { apiStore } from '@/plugins/store'
import router from '@/router'
import Cookies from 'js-cookie'

axios.interceptors.response.use(null, error => {
  if (error.status === 401) {
    logout().then(() => {})
  }
  return Promise.reject(error)
})

function getJWTPayloadFromCookie () {
  const jwtHeaderAndPayload = Cookies.get('jwt_hp')
  if (!jwtHeaderAndPayload) return ''

  return jwtHeaderAndPayload.split('.')[1]
}

function parseJWTPayload (payload) {
  if (!payload) return {}
  const base64 = payload.replace(/-/g, '+').replace(/_/g, '/')
  const jsonPayload = decodeURIComponent(atob(base64).split('').map(function (c) {
    return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
  }).join(''))

  return JSON.parse(jsonPayload)
}

function getJWTExpirationTimestamp () {
  return (parseJWTPayload(getJWTPayloadFromCookie()).exp ?? 0) * 1000
}

export function isLoggedIn () {
  return Date.now() < getJWTExpirationTimestamp()
}

async function login (username, password) {
  const url = await apiStore.href(apiStore.get(), 'login')
  return apiStore.post(url, { username: username, password: password }).then(() => {
    return isLoggedIn()
  })
}

async function resetPasswordRequest (email, token) {
  const url = await apiStore.href(apiStore.get(), 'resetPassword')
  return apiStore.post(url, { email: email, token: token })
}

async function resetPassword (id, password, token) {
  const url = await apiStore.href(apiStore.get(), 'resetPassword', { id: id })
  return apiStore.patch(url, { password: password, token: token })
}

function user () {
  if (!getJWTPayloadFromCookie()) {
    return null
  }
  const user = apiStore.get(parseJWTPayload(getJWTPayloadFromCookie()).user)
  user._meta.load = user._meta.load.catch(e => {
    if (e.response && [401, 403, 404].includes(e.response.status)) {
      // 401 means no complete token was submitted, so we may be missing the JWT signature cookie
      // 403 means we can theoretically interact in some way with the user, but apparently not read it
      // 404 means the user doesn't exist or we don't have access to it
      // Either way, we aren't allowed to access the user from the token, so it's best to ask the user
      // to log in again.
      auth.logout()
      return
    }
    throw e
  })
  return user
}

async function register (data) {
  const url = await apiStore.href(apiStore.get(), 'users')
  return apiStore.post(url, data)
}

async function redirectToOAuthLogin (provider) {
  let returnUrl = window.location.origin + router.resolve({ name: 'loginCallback' }).href

  const params = new URLSearchParams(window.location.search)
  if (params.has('redirect')) {
    returnUrl += '?redirect=' + params.get('redirect')
  }

  return apiStore.href(apiStore.get(), provider, { callback: encodeURI(returnUrl) }).then(url => {
    window.location.href = window.environment.API_ROOT_URL + url
  })
}

async function loginGoogle () {
  return redirectToOAuthLogin('oauthGoogle')
}

async function loginPbsMiData () {
  return redirectToOAuthLogin('oauthPbsmidata')
}

async function loginCeviDB () {
  return redirectToOAuthLogin('oauthCevidb')
}

export async function logout () {
  Cookies.remove('jwt_hp', { domain: window.environment.SHARED_COOKIE_DOMAIN })
  return router.push({ name: 'login' })
    .then(() => apiStore.purgeAll())
    .then(() => isLoggedIn())
}

export const auth = {
  isLoggedIn,
  login,
  register,
  loginGoogle,
  loginPbsMiData,
  loginCeviDB,
  logout,
  user,
  resetPasswordRequest,
  resetPassword
}

class AuthPlugin {
  install (Vue) {
    Object.defineProperties(Vue.prototype, {
      $auth: {
        get () {
          return auth
        }
      }
    })
  }
}

export default new AuthPlugin()
