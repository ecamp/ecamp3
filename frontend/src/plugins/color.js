import { ColorSpace, sRGB } from 'colorjs.io/fn'

export default {
  install: () => {
    ColorSpace.register(sRGB)
  },
}
