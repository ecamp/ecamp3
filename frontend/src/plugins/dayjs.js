import dayjs from '@/../../common/helpers/dayjs.js'

export default {
  install: (Vue) => {
    Object.defineProperties(Vue.prototype, {
      $date: {
        get () {
          return dayjs
        }
      }
    })
    Vue.dayjs = dayjs
  }
}
