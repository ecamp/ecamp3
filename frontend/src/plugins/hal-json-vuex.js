import HalJsonVuex from 'hal-json-vuex'
import axios from 'axios'
import { createStore } from 'vuex'
import { getEnv } from '@/environment.js'

export default {
  install: (app, options) => {
    // create store
    const store = createStore({
      state() {
        return {}
      },
    })
    app.use(store)

    const axiosInstance = axios.create({
      withCredentials: true,
      baseURL: getEnv().API_ROOT_URL,
      headers: { common: { Accept: 'application/hal+json' } },
    })
    axiosInstance.interceptors.request.use(function (config) {
      if (config.method === 'patch') {
        config.headers['Content-Type'] = 'application/merge-patch+json'
      }
      return config
    })

    // create and inject API
    const api = new HalJsonVuex(store, axiosInstance, {
      forceRequestedSelfLink: true,
    })

    app.use(api)
  },
}
