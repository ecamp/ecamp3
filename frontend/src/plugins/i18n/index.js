import { createI18n } from 'vue-i18n'
import deepmerge from 'deepmerge'

import itCommon from '@/common/locales/it.json'
import itCHScoutCommon from '@/common/locales/it-CH-scout.json'
import frCommon from '@/common/locales/fr.json'
import frCHScoutCommon from '@/common/locales/fr-CH-scout.json'
import enCommon from '@/common/locales/en.json'
import enCHScoutCommon from '@/common/locales/en-CH-scout.json'
import deCommon from '@/common/locales/de.json'
import deCHScoutCommon from '@/common/locales/de-CH-scout.json'
import rmCommon from '@/common/locales/rm.json'
import rmCHScoutCommon from '@/common/locales/rm-CH-scout.json'

import it from '@/locales/it.json'
import itCHScout from '@/locales/it-CH-scout.json'
import fr from '@/locales/fr.json'
import frCHScout from '@/locales/fr-CH-scout.json'
import en from '@/locales/en.json'
import enCHScout from '@/locales/en-CH-scout.json'
import de from '@/locales/de.json'
import deCHScout from '@/locales/de-CH-scout.json'
import rm from '@/locales/rm.json'
import rmCHScout from '@/locales/rm-CH-scout.json'

import validationIt from '@vee-validate/i18n/dist/locale/it.json'
import validationFr from '@vee-validate/i18n/dist/locale/fr.json'
import validationEn from '@vee-validate/i18n/dist/locale/en.json'
import validationDe from '@vee-validate/i18n/dist/locale/de.json'

import vuetifyEn from 'vuetify/lib/locale/en'
import vuetifyDe from 'vuetify/lib/locale/de'
import vuetifyFr from 'vuetify/lib/locale/fr'
import vuetifyIt from 'vuetify/lib/locale/it'

const fallbackLocales = {
  rm: ['de'],
  default: 'en',
}

const i18n = createI18n({
  locale: 'de',
  fallbackLocale: fallbackLocales,
  messages: deepmerge.all([
    // vee-validate locales
    {
      it: {
        global: {
          validation: validationIt.messages,
        },
      },
      fr: {
        global: {
          validation: validationFr.messages,
        },
      },
      en: {
        global: {
          validation: validationEn.messages,
        },
      },
      de: {
        global: {
          validation: validationDe.messages,
        },
      },
    },

    // vuetify locales
    {
      en: { $vuetify: vuetifyEn },
      de: { $vuetify: vuetifyDe },
      fr: { $vuetify: vuetifyFr },
      it: { $vuetify: vuetifyIt },
    },

    // eCamp common locales
    {
      it: itCommon,
      'it-CH-scout': itCHScoutCommon,
      fr: frCommon,
      'fr-CH-scout': frCHScoutCommon,
      en: enCommon,
      'en-CH-scout': enCHScoutCommon,
      de: deCommon,
      'de-CH-scout': deCHScoutCommon,
      rm: rmCommon,
      'rm-CH-scout': rmCHScoutCommon,
    },

    // eCamp frontend only locales
    {
      it,
      'it-CH-scout': itCHScout,
      fr,
      'fr-CH-scout': frCHScout,
      en,
      'en-CH-scout': enCHScout,
      de,
      'de-CH-scout': deCHScout,
      rm,
      'rm-CH-scout': rmCHScout,
    },
  ]),
  silentTranslationWarn: true,
})

// Vue.use(i18n)

Object.defineProperty(i18n, 'browserPreferredLocale', {
  get: function () {
    const languages = navigator.languages || [navigator.language]
    for (const language of languages) {
      if (this.availableLocales.includes(language)) {
        return language
      }

      const languageFallback = language.substring(0, 2)
      if (this.availableLocales.includes(languageFallback)) {
        return languageFallback
      }
    }
    return fallbackLocales.default
  },
})

export default i18n

export { i18n, fallbackLocales }
