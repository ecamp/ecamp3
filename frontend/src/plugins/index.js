export { default as vuetify } from './vuetify'
export { default as store } from './store'
export { default as apiStore } from './store/apiPlugin'
export { default as auth } from './auth'
export { default as filterLoading } from './filterLoading'
export { default as formBaseComponents } from './formBaseComponents'
export { default as ignoreNativeBindingWarnMessages } from './ignoreNativeBindingWarnMessages'
export { default as i18n } from './i18n'
export { default as veeValidate } from './veeValidate'
export { vueMoment, moment } from './vueMoment'
