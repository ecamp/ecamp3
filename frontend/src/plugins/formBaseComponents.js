// Global registration of all Vue components in folder components/form/base

import ECheckbox from '@/components/form/base/ECheckbox.vue'
import EColorPicker from '@/components/form/base/EColorPicker.vue'
import EDatePicker from '@/components/form/base/EDatePicker.vue'
import EForm from '@/components/form/base/EForm.vue'
import ERichtext from '@/components/form/base/ERichtext.vue'
import ESelect from '@/components/form/base/ESelect.vue'
import ESwitch from '@/components/form/base/ESwitch.vue'
import ETextarea from '@/components/form/base/ETextarea.vue'
import ETextField from '@/components/form/base/ETextField.vue'
import ETimePicker from '@/components/form/base/ETimePicker.vue'

class FormBaseComponentsPlugin {
  install(Vue) {
    Vue.component('ECheckbox', ECheckbox)
    Vue.component('EColorPicker', EColorPicker)
    Vue.component('EDatePicker', EDatePicker)
    Vue.component('EForm', EForm)
    Vue.component('ERichtext', ERichtext)
    Vue.component('ESelect', ESelect)
    Vue.component('ESwitch', ESwitch)
    Vue.component('ETextarea', ETextarea)
    Vue.component('ETextField', ETextField)
    Vue.component('ETimePicker', ETimePicker)
  }
}

export default new FormBaseComponentsPlugin()
