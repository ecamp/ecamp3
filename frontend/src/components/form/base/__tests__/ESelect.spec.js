import Vue from 'vue'
import Vuetify from 'vuetify'

import { formBaseComponents } from '@/plugins'

import { mount as mountComponent } from '@vue/test-utils'
import ESelect from '../ESelect.vue'

Vue.use(Vuetify)
Vue.use(formBaseComponents)

describe('An ESelect', () => {
  let vuetify

  const FIRST_OPTION = {
    value: 1,
    text: 'firstOption',
  }
  const SECOND_OPTION = {
    value: '2',
    text: 'secondOption',
  }
  const THIRD_OPTION = {
    value: { key: 'value', array: [1, 2, 3], nestedObject: { key: 'value' } },
    text: 'thirdOption',
  }
  const selectValues = [FIRST_OPTION, SECOND_OPTION, THIRD_OPTION]

  const mount = (options) => {
    const app = Vue.component('App', {
      components: { ESelect },
      data: function () {
        return {
          selectValues: selectValues,
          data: null,
        }
      },
      template: `
          <div data-app>
          <e-select :items="selectValues" v-model="data"/>
          </div>
        `,
    })
    return mountComponent(app, { vuetify, attachTo: document.body, ...options })
  }

  beforeEach(() => {
    vuetify = new Vuetify()
  })
  test('looks like a dropdown', async () => {
    const wrapper = mount()
    expect(wrapper).toMatchSnapshot('no item selected')

    await wrapper.find('.v-input__slot').trigger('click')
    expect(wrapper).toMatchSnapshot('dropdown open')

    await wrapper.findAll('[role="option"]').at(0).trigger('click')
    expect(wrapper).toMatchSnapshot('dropdown closed with selected value')

    await wrapper.find('.v-input__slot').trigger('click')
    expect(wrapper).toMatchSnapshot('dropdown open with selected value')
  })

  test('update viewmodel with selected value', async () => {
    const wrapper = mount()
    expect(wrapper.vm.data).toBeNull()

    await wrapper.find('.v-input__slot').trigger('click')
    await wrapper.findAll('[role="option"]').at(0).trigger('click')
    expect(wrapper.vm.data).toBe(FIRST_OPTION.value)

    await wrapper.find('.v-input__slot').trigger('click')
    await wrapper.findAll('[role="option"]').at(2).trigger('click')
    expect(wrapper.vm.data).toBe(THIRD_OPTION.value)
  })

  test('update selected value with viewmodel', async () => {
    const wrapper = mount()

    await wrapper.setData({ data: SECOND_OPTION.value })
    expect(wrapper.html()).toContain(SECOND_OPTION.text)
    expect(wrapper.html()).not.toContain(FIRST_OPTION.text)

    await wrapper.setData({ data: FIRST_OPTION.value })
    expect(wrapper.html()).toContain(FIRST_OPTION.text)
    expect(wrapper.html()).not.toContain(SECOND_OPTION.text)
  })

  test('update selected value after it was open', async () => {
    const wrapper = mount()

    await wrapper.find('.v-input__slot').trigger('click')
    await wrapper.findAll('[role="option"]').at(0).trigger('click')
    expect(wrapper.vm.data).toBe(FIRST_OPTION.value)

    await wrapper.setData({ data: SECOND_OPTION.value })
    expect(
      wrapper.findAll('[role="option"]').at(1).element.getAttribute('aria-selected')
    ).toBe('true')
    expect(
      wrapper.findAll('[role="option"]').at(0).element.getAttribute('aria-selected')
    ).not.toBe('true')
  })
})
