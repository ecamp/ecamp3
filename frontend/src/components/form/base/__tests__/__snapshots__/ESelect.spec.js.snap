// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`An ESelect > looks like a dropdown > dropdown closed with selected value 1`] = `
<div data-app>
     <div class="e-form-container"
          items="[object Object],[object Object],[object Object]"
          value="1">
          <div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty v-input--is-focused theme--light v-text-field v-text-field--single-line v-text-field--filled v-text-field--enclosed v-select primary--text">
               <div class="v-input__control">
                    <div aria-expanded="false"
                         aria-haspopup="listbox"
                         aria-owns="list-6"
                         class="v-input__slot"
                         role="button">
                         <div class="v-select__slot">
                              <div class="v-select__selections">
                                   <div class="v-select__selection v-select__selection--comma">firstOption</div>
                                   <input aria-readonly="false"
                                          autocomplete="off"
                                          id
                                          readonly="readonly"
                                          type="text">
                              </div>
                              <div class="v-input__append-inner">
                                   <div class="v-input__icon v-input__icon--append">
                                        <i aria-hidden="true"
                                           class="v-icon notranslate mdi mdi-menu-down theme--light primary--text"></i>
                                   </div>
                              </div>
                              <input type="hidden"
                                     value="1">
                         </div>
                         <div class="v-menu">
                              <transition-stub name="v-menu-transition"></transition-stub>
                         </div>
                    </div>
               </div>
          </div>
     </div>
     <div class="v-menu__content theme--light "
          style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 8; display: none;">
          <div class="v-list v-select-list v-sheet theme--light theme--light"
               id
               role="listbox"
               tabindex="-1">
               <div aria-selected="true"
                    class="v-list-item primary--text v-list-item--active v-list-item--link theme--light"
                    id
                    role="option"
                    tabindex="0">
                    <div class="v-list-item__content">
                         <div class="v-list-item__title">firstOption</div>
                    </div>
               </div>
               <div aria-selected="false"
                    class="v-list-item v-list-item--link theme--light"
                    id
                    role="option"
                    tabindex="0">
                    <div class="v-list-item__content">
                         <div class="v-list-item__title">secondOption</div>
                    </div>
               </div>
               <div aria-selected="false"
                    class="v-list-item v-list-item--link theme--light"
                    id
                    role="option"
                    tabindex="0">
                    <div class="v-list-item__content">
                         <div class="v-list-item__title">thirdOption</div>
                    </div>
               </div>
          </div>
     </div>
</div>
`;

exports[`An ESelect > looks like a dropdown > dropdown open 1`] = `
<div data-app>
     <div class="e-form-container"
          items="[object Object],[object Object],[object Object]">
          <div class="v-input v-input--hide-details v-input--is-focused theme--light v-text-field v-text-field--single-line v-text-field--filled v-text-field--enclosed v-select v-select--is-menu-active primary--text">
               <div class="v-input__control">
                    <div aria-expanded="true"
                         aria-haspopup="listbox"
                         aria-owns="list-6"
                         class="v-input__slot"
                         role="button">
                         <div class="v-select__slot">
                              <div class="v-select__selections">
                                   <input aria-readonly="false"
                                          autocomplete="off"
                                          id
                                          readonly="readonly"
                                          type="text">
                              </div>
                              <div class="v-input__append-inner">
                                   <div class="v-input__icon v-input__icon--append">
                                        <i aria-hidden="true"
                                           class="v-icon notranslate mdi mdi-menu-down theme--light primary--text"></i>
                                   </div>
                              </div>
                              <input type="hidden">
                         </div>
                         <div class="v-menu">
                              <transition-stub name="v-menu-transition"></transition-stub>
                         </div>
                    </div>
               </div>
          </div>
     </div>
     <div class="v-menu__content theme--light menuable__content__active "
          style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 8; display: none;">
          <div class="v-list v-select-list v-sheet theme--light theme--light"
               id
               role="listbox"
               tabindex="-1">
               <div aria-selected="false"
                    class="v-list-item v-list-item--link theme--light"
                    id
                    role="option"
                    tabindex="0">
                    <div class="v-list-item__content">
                         <div class="v-list-item__title">firstOption</div>
                    </div>
               </div>
               <div aria-selected="false"
                    class="v-list-item v-list-item--link theme--light"
                    id
                    role="option"
                    tabindex="0">
                    <div class="v-list-item__content">
                         <div class="v-list-item__title">secondOption</div>
                    </div>
               </div>
               <div aria-selected="false"
                    class="v-list-item v-list-item--link theme--light"
                    id
                    role="option"
                    tabindex="0">
                    <div class="v-list-item__content">
                         <div class="v-list-item__title">thirdOption</div>
                    </div>
               </div>
          </div>
     </div>
</div>
`;

exports[`An ESelect > looks like a dropdown > dropdown open with selected value 1`] = `
<div data-app>
     <div class="e-form-container"
          items="[object Object],[object Object],[object Object]"
          value="1">
          <div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty v-input--is-focused theme--light v-text-field v-text-field--single-line v-text-field--filled v-text-field--enclosed v-select v-select--is-menu-active primary--text">
               <div class="v-input__control">
                    <div aria-expanded="true"
                         aria-haspopup="listbox"
                         aria-owns="list-6"
                         class="v-input__slot"
                         role="button">
                         <div class="v-select__slot">
                              <div class="v-select__selections">
                                   <div class="v-select__selection v-select__selection--comma">firstOption</div>
                                   <input aria-readonly="false"
                                          autocomplete="off"
                                          id
                                          readonly="readonly"
                                          type="text">
                              </div>
                              <div class="v-input__append-inner">
                                   <div class="v-input__icon v-input__icon--append">
                                        <i aria-hidden="true"
                                           class="v-icon notranslate mdi mdi-menu-down theme--light primary--text"></i>
                                   </div>
                              </div>
                              <input type="hidden"
                                     value="1">
                         </div>
                         <div class="v-menu">
                              <transition-stub name="v-menu-transition"></transition-stub>
                         </div>
                    </div>
               </div>
          </div>
     </div>
     <div class="v-menu__content theme--light menuable__content__active "
          style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 8; display: none;">
          <div class="v-list v-select-list v-sheet theme--light theme--light"
               id
               role="listbox"
               tabindex="-1">
               <div aria-selected="true"
                    class="v-list-item primary--text v-list-item--active v-list-item--link theme--light"
                    id
                    role="option"
                    tabindex="0">
                    <div class="v-list-item__content">
                         <div class="v-list-item__title">firstOption</div>
                    </div>
               </div>
               <div aria-selected="false"
                    class="v-list-item v-list-item--link theme--light"
                    id
                    role="option"
                    tabindex="0">
                    <div class="v-list-item__content">
                         <div class="v-list-item__title">secondOption</div>
                    </div>
               </div>
               <div aria-selected="false"
                    class="v-list-item v-list-item--link theme--light"
                    id
                    role="option"
                    tabindex="0">
                    <div class="v-list-item__content">
                         <div class="v-list-item__title">thirdOption</div>
                    </div>
               </div>
          </div>
     </div>
</div>
`;

exports[`An ESelect > looks like a dropdown > no item selected 1`] = `
<div data-app>
     <div class="e-form-container"
          items="[object Object],[object Object],[object Object]">
          <div class="v-input v-input--hide-details theme--light v-text-field v-text-field--single-line v-text-field--filled v-text-field--enclosed v-select">
               <div class="v-input__control">
                    <div aria-expanded="false"
                         aria-haspopup="listbox"
                         aria-owns="list-6"
                         class="v-input__slot"
                         role="button">
                         <div class="v-select__slot">
                              <div class="v-select__selections">
                                   <input aria-readonly="false"
                                          autocomplete="off"
                                          id
                                          readonly="readonly"
                                          type="text">
                              </div>
                              <div class="v-input__append-inner">
                                   <div class="v-input__icon v-input__icon--append">
                                        <i aria-hidden="true"
                                           class="v-icon notranslate mdi mdi-menu-down theme--light"></i>
                                   </div>
                              </div>
                              <input type="hidden">
                         </div>
                         <div class="v-menu">
                              <!---->
                         </div>
                    </div>
               </div>
          </div>
     </div>
</div>
`;
